### ISSUE TITLE
Hexadecimal/Octal support for enums

### ISSUE THREAD

**Issue by aprokop** [sethrj/swig#33]
If we have a c++ header with, for example, the following:
```c++
enum StatusType {
  Passed = 0x1,
  Failed = 0x2,
  Undefined = 0x4
};
```
it results in an error on the Fortran side:
```Fortran
 enum, bind(c)
  enumerator :: StatusType = -1
  enumerator :: Passed = 0x1
  enumerator :: Failed = 0x2
  enumerator :: Undefined = 0x4
 end enum
```
as Fortran does not support `0x` hex notations. The values should be converted to Fortran types. An example of how it could be done can be seen in `python.cxx` function`convertValue` (around line 2136).

**Comment by sethrj** [comment]
I'm looking into this now using the `enum_thorough` c++ test: `cd Examples/test-suite/fortran && make enum_thorough.cpptest`.

Since the value of the enum can be *any* valid compile-time expression (e.g. the test has `(('A' << 24)|('B' << 16)|('C' << 8)|'D')`), I don't know that we can reasonably create compile-time enums in Fortran. (An alternative may be to declare them as external ints in the C code? That way the C compiler can evaluate the constants.)

**Comment by aprokop** [comment]
I don't know what the long term solution should be. But for now, ForTrilinos needs just regular `0x...` constants from `BelosTypes.hpp` file.

**Comment by sethrj** [comment]
OK, I think I have a solution. Instead of using Fortran enumerators (or maybe provide a feature to use native enums):
```
integer(kind = c_int), protected, &
    bind(c, name='MyEnum_GREEN') :: GREEN
integer(kind = c_int), protected, &
    bind(c, name='MyEnum_BLUE') :: BLUE
```
with C wrapper code:
```
int MyEnum_GREEN = GREEN;
int MyEnum_BLUE = BLUE;
```

**Comment by aprokop** [comment]
@mattbement What's your take on this?

**Comment by sethrj** [comment]
@aprokop Implemented in b1269d613913effeef08a5ffb204c0da2225e2a0
